### PR TITLE
INTMDB-528-Feature custom dns config cluster aws

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: cfn-resources/custom-dns-configuration-cluster-aws/cmd/resource/resource.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progress_events "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+var CreateRequiredFields = []string{constants.ProjectID, constants.PubKey, constants.PvtKey}
+var ReadRequiredFields = []string{constants.ProjectID, constants.PubKey, constants.PvtKey}
+var UpdateRequiredFields []string
```
Comment (rdharmedran) [inline cfn-resources/custom-dns-configuration-cluster-aws/cmd/resource/resource.go:20]: use one array variable if all the input fields are same for create,read,delete,list

File: cfn-resources/custom-dns-configuration-cluster-aws/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# MongoDB::Atlas::CustomDnsConfigurationClusterAws
+
+## Description
+Returns, adds and removes custom dns configuration for AWS cluster.
```
Comment (Zuhairahmed) [inline cfn-resources/custom-dns-configuration-cluster-aws/README.md:4]:
```suggestion
Returns, adds and removes custom DNS configurations for MongoDB Atlas database deployments on AWS.
```

File: cfn-resources/custom-dns-configuration-cluster-aws/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+|-----------|------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| PATCH     | true/false | [From MongoDB](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Custom-DNS-for-Atlas-Clusters-Deployed-to-AWS/operation/toggleOneStateOfOneCustomDnsConfigurationForAtlasClustersOnAws) |
+| READ      | true/false | [From MongoDB](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Custom-DNS-for-Atlas-Clusters-Deployed-to-AWS)                                    |
+
+
+For more information see: MongoDB Atlas API [CustomeDNSConfigurationClusterAWS Endpoint](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Custom-DNS-for-Atlas-Clusters-Deployed-to-AWS) Documentation.
```
Comment (Zuhairahmed) [inline cfn-resources/custom-dns-configuration-cluster-aws/README.md:64]:
```suggestion
For more information see: MongoDB Atlas API Endpoint [Custom DNS for Atlas Clusters Deployed to AWS](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Custom-DNS-for-Atlas-Clusters-Deployed-to-AWS) Documentation.
```

File: cfn-resources/custom-dns-configuration-cluster-aws/README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+cd ${repo_root}/cfn-resources/custom-dns-configuration-cluster-aws
+./test/custom-dns-config-cluster-aws.create-sample-cfn-request.sh > test.request.json
+echo "Sample request:"
+cat test.request.json
+cfn invoke CREATE test.request.json
+cfn invoke DELETE test.request.json
+```
+
+Both CREATE & DELETE tests must pass.
+
```
Comment (Zuhairahmed) [inline cfn-resources/custom-dns-configuration-cluster-aws/README.md:36]: Missing Example Section, can you add and link to Example directory: https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples

File: examples/custom-dns-configuration-cluster-aws/customDnsConfigurationClusterAws.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
+  "AWSTemplateFormatVersion" : "2010-09-09",
+  "Description" : "This template creates a Project on the MongoDB Atlas API, this will be billed to your Atlas account.",
```
Comment (Zuhairahmed) [inline examples/custom-dns-configuration-cluster-aws/customDnsConfigurationClusterAws.json:3]: This looks like Example for Project Resource. Can you refresh for custom DNS Configuration Cluster AWS

**Review by Zuhairahmed — CHANGES_REQUESTED** [review]
Few comments, can you resolve and then ping me on Slack to review again when ready. thanks and great work here @govardhanpagidi!

**Comment by Zuhairahmed** [comment]
Missing GitHub Build + Linting Error checks, please add

**Comment by ParthasarathyVaradhan** [comment]
Build and Lint Checks are happening.
https://github.com/mongodb/mongodbatlas-cloudformation-resources/actions/runs/3742624161/jobs/6353800277
https://github.com/mongodb/mongodbatlas-cloudformation-resources/actions/runs/3742624165/jobs/6353800293

File: cfn-resources/custom-dns-configuration-cluster-aws/README.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+## CloudFormation Examples
```
Comment (Zuhairahmed) [inline cfn-resources/custom-dns-configuration-cluster-aws/README.md:37]:
```suggestion
## CloudFormation Example
```

File: cfn-resources/custom-dns-configuration-cluster-aws/README.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+Please see the [CFN Template](test/custom-dns-config-cluster-aws.sample-cfn-request.json) for example resource
```
Comment (Zuhairahmed) [inline cfn-resources/custom-dns-configuration-cluster-aws/README.md:39]:
```suggestion
Please see the [custom-dns-configuration-cluster-aws](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples/custom-dns-configuration-cluster-aws) for example resource
```

**Review by Zuhairahmed — APPROVED** [review]
Few minor comments, please fix. Not blocking. Otherwise LGTM! thanks @govardhanpagidi

**Review by ParthasarathyVaradhan — APPROVED** [review]
LGTM. Build file is correct for Multi region deployment

**Review by swattyT — APPROVED** [review]
LGTM!

**Pull request merged**